src/Simulation/TargetDefinitions/Decompositions/Measure.qs

```diff
@@ -40,29 +41,115 @@ namespace Microsoft.Quantum.Intrinsic {
     /// operation will fail.
     operation Measure (bases : Pauli[], qubits : Qubit[]) : Result {
         if (Length(bases) != Length(qubits)) { fail "Arrays 'bases' and 'qubits' must be of the same length."; }
-        mutable res = One;
-        if( Length(bases) == 1 ) {
+        if Length(bases) == 0 {
+            return Zero;
+        }
+        elif Length(bases) == 1 {
+            MapPauli(qubits[0], PauliZ, bases[0]);
+            let res = MZ(qubits[0]);
+            Adjoint MapPauli(qubits[0], PauliZ, bases[0]);
+            return res;
+        }
+        elif Length(bases) == 2 {
+            use q = Qubit();
             within {
-                MapPauli(qubits[0], PauliZ, bases[0]);
+                H(q);
             }
             apply {
-                set res = M(qubits[0]);
-                PreparePostM(res, qubits[0]);
+                JointMeasureHelper(bases[0], q, qubits[0]);
```

Comment:
I'll readily admit this is a nitpick, but `JointMeasureHelper` isn't a verb as suggested for Q# operation names.

Reply:
I updated the name to `EntangleForJointMeasurement` to match style guide.